Experimental/CameraComponent/FirstPersonCamera.cs

```diff
@@ -1,264 +1,166 @@
+                float curr;
+                _propertySet.TryGetScalar("Roll", out curr);
+                return curr;
             }
             set
             {
-                if (UseAnimations)
-                {
-                    propertySet.InsertScalar("Roll", value);
-                }
-                else
-                {
-                    roll = value;
-                    RaisePropertyChanged("Roll");
-                }
+                _propertySet.InsertScalar("Roll", value);
             }
         }
+
+        // Camera's location in world space
         public Vector3 Position
         {
             get
             {
-                if(UseAnimations)
-                {
-                    Vector3 curr = Vector3.Zero;
-                    return (propertySet.TryGetVector3("Position", out curr) == CompositionGetValueStatus.Succeeded) ? curr : position;
-                }
-                else
-                {
-                    return position;
-                }
+                Vector3 curr;
+                _propertySet.TryGetVector3("Position", out curr);
+                return curr;
             }
             set
             {
-                if (UseAnimations)
-                {
-                    propertySet.InsertVector3("Position", value);
-                }
-                else
-                {
-                    position = value;
-                    RaisePropertyChanged("Position");
-                }
+                _propertySet.InsertVector3("Position", value);
             }
         }
-        public Vector3 LookDirection
+
+        // The camera's projection, orthographic or perspective
+        public Projection Projection
         {
-            get => Vector3.Zero;
+            get => _projection;
             set
             {
-                Vector3 direction = value;
-                if (value != Vector3.Zero)
+                _projection = value;
+
+                // create view matrix based on the camera's rotation and position
+                var matPos = "Matrix4x4.CreateTranslation(-FPCamera.Position)";
+                var matRoll = "Matrix4x4.CreateFromAxisAngle(Vector3(0, 0, 1), -FPCamera.Roll)";
+                var matPitch = "Matrix4x4.CreateFromAxisAngle(Vector3(1, 0, 0), -FPCamera.Pitch)";
+                var matYaw = "Matrix4x4.CreateFromAxisAngle(Vector3(0, 1, 0), -FPCamera.Yaw)";
+                var viewMat = matPos + "*" + matYaw + "*" + matPitch + "*" + matRoll;
+
+                // create a matrix that is the product of the camera's view matrix and the projection's projection matrix
+                var modelViewProjMatExpression = _compositor.CreateExpressionAnimation();
+                if (_projection == null) // if null then the default is an orthographic projection
                 {
-                    direction = Vector3.Normalize(value);
+                    OrthographicProjection defaultProj = new OrthographicProjection();

```

Comment:
The default orthographic projection in your OrthographicProjection class is 1000x1000.  Our Scenes API has a concept of a "default camera" which is an orthographic projection of 1x1, but of course that has no stretch properties.  Should we be mimicking the default camera projection?

Reply:
Ideally the default would be Orthographic 1x1 with Stretch.None. I think the problem with this though is if we implemented it when the user sets a Projection to null then we would be allowing the camera to reach into the viewport and change its stretch. Moreover, if we are not going to allow the camera to change the viewport's stretch property, then 1x1 for an ortho projection looks pretty terrible.

Reply:
I think a Camera with a null projection should either be invalid or give the "default camera" behavior.  This goes for a Viewport with a null Camera as well.